modules/api/src/main/java/org/apache/ignite/lang/NoMorePagesException.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.lang;
+
+import static org.apache.ignite.lang.ErrorGroups.Common.CURSOR_NO_MORE_PAGES_ERR;
+
+/**
+ * Exception is thrown when there are no more pages in the cursor.
+ */
+public class NoMorePagesException extends IgniteException {
```

Comment:
Seems, this class is not used.

Reply:
Restored the use of `NoMorePagesException` since `mapToPublicSqlException` will wrap `NoSuchElementException` into `IgniteException`

Reply:
1. Do we really need this exception? 
It means a cursor reach end of data and can be closed. Maybe, we could close it instantly and throw CursorClosedException. Is there any case, when user need to know the real reason?

2. "Pages" in NoMorePagesException name looks a bit inconsistent with the method signature `util/AsyncCursor`.
It returns "batch", not a page.
`CompletableFuture<BatchedResult<T>> requestNextAsync(int rows);`
 However, public interface `lang/AsyncCursor` have different method.
`CompletableFuture<? extends AsyncCursor<T>> fetchNextPage();`

Reply:
> 1. On the other hand NoMorePagesException is a valid case, and you can complete the processing.
Cursor has methods `hasMorePages` or there is a flag in BatchedResult. So, NoMorePagesException will be thrown only if the one violates the contract.